homeassistant/components/switch/switchmate.py

```diff
@@ -43,17 +43,42 @@ class Switchmate(SwitchDevice):
 
     def __init__(self, mac, name) -> None:
         """Initialize the Switchmate."""
-        # pylint: disable=import-error
-        import bluepy
         self._state = False
         self._name = name
         self._mac = mac
+        self._device = None
+        if not self._connect():
+            raise PlatformNotReady()
+
+    def _connect(self) -> bool:
+        # pylint: disable=import-error
+        import bluepy
+        if self._device:
+            try:
+                self._device.disconnect()
+            except bluepy.btle.BTLEException:
+                pass
         try:
             self._device = bluepy.btle.Peripheral(self._mac,
                                                   bluepy.btle.ADDR_TYPE_RANDOM)
         except bluepy.btle.BTLEException:
-            _LOGGER.error("Failed to set up switchmate")
-            raise PlatformNotReady()
+            _LOGGER.error("Failed to connect to switchmate")
+            return False
+        return True
+
+    def _sendpacket(self, key, retry=2) -> bool:
```

Comment:
This is what we regularly ask to be in a standalone library.

Reply:
Ok, I thought it might be ok since it so few lines.
But I will make a standalone library.